developer/src/kmldmlc/src/keyman/kmx/kmx-plus-builder.ts

```diff
@@ -0,0 +1,319 @@
+import * as r from 'restructure';
+import KMXPlusFile from "./kmx-plus";
+import { constants } from '@keymanapp/ldml-keyboard-constants';
+
+interface BUILDER_SECTION {
+  ident: number;
+  size: number;
+  _offset: number; // used only for building the output
+};
+
+interface BUILDER_SECT_ITEM {
+  sect: number;
+  offset: number; //? new r.VoidPointer(r.uint32le, {type: 'global'})
+};
+
+interface BUILDER_SECT extends BUILDER_SECTION {
```

Comment:
I may be a little lost here but is this interface only ever used for TOC  could we call it `BUILDER_TOC`.  The name `BUILDER_SECTION` and `BUILDER_SECT` are similar. Just a readability comment.

Reply:
`SECT` matches the section name of `sect`. It is a little confusing but not sure what else to do with it at present!

Reply:
I realized that also.  `sect` is the array of SECTion offsets. 

Could add an extraneous comment just to clarify

```suggestion
// Builder for the 'sect' (Section table of contents) section
interface BUILDER_SECT extends BUILDER_SECTION {
```